Simplenote/CoreDataManager.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import CoreData
+
+@objcMembers
+class CoreDataManager: NSObject {
+
+    /// URL for the managed object model resource
+    ///
+    static private let modelURL: URL? = {
+        guard let path = Bundle.main.path(forResource: Constants.resourceName, ofType: Constants.resourceType) else {
+            return nil
+        }
+        return URL(string: path)
+    }()
+
+    /// URL for the in app documents directory
+    ///
+    static let documentsDirectory: URL = {
+        FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+    }()
+
+    /// In app core data storge URL
+    ///
+    static let appStorageURL: URL = {
+        documentsDirectory.appendingPathComponent(Constants.sqlFile)
+    }()
+
+    /// URL for Simplenote's shared app group directory
+    ///
+    static let groupDirectory: URL = {
+        FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: Constants.sharedDirectoryDomain + Constants.groupIdentifier)!
+    }()
+
+    /// URL for Simplenote's shared app group documents directory
+    ///
+    static let groupDocumentsDirectory: URL = {
+        groupDirectory.appendingPathComponent(Constants.documentDirectory)
+    }()
+
+    /// URL for core data storage in shared app group documents directory
+    ///
+    static let groupStorageURL: URL = {
+        groupDocumentsDirectory.appendingPathComponent(Constants.sqlFile)
+    }()
+
+    /// Bool checking if the in app database exsists
+    ///
+    static let oldDbExists: Bool = {
+        FileManager.default.fileExists(atPath: CoreDataManager.appStorageURL.path)
+    }()
+
+    /// Bool checking if the app group database exsists
+    ///
+    static let appGroupDbExists: Bool = {
+        FileManager.default.fileExists(atPath: CoreDataManager.groupStorageURL.path)
+    }()
```

Comment:
`oldDbExists` + `appGroupDbExists` should not be `static let` (since after the migration they'll preserve their old / incorrect value).

How about something like:

```
var storageURL: URL {
    // Logic to return the actual StorageURL we need    
}

private var legacyStorageURL: URL { }
private var sharedStorageURL: URL { }
```

And drop both `oldDbExists` + `appGroupDbExists`?